:book: Document our compatibility notices for sub-folders (#3579)
kubernetes-sigs/cluster-api: test/README.md

# Test

## Compatibility notice

> Code under the `test/` subfolder is not subject to deprecation notices or compatbility guarantees.

Reviewer: ```suggestion
This package is not subject to deprecation notices or compatibility guarantees.
```

> - The test framework for Cluster API, given its quick pace allows for breaking changes, external providers using this codebase should update to the latest API changes when updating Cluster API. Maintainers and contributors must give notice in release notes when a breaking change happens.

Reviewer: ```suggestion
- We iterate on the test framework quickly and frequently, and breaking changes are likely. External providers using this package should update to the latest API changes when updating Cluster API. Maintainers and contributors must give notice in release notes when a breaking change happens.
```

> - The docker provider, including its APIs should only be used in development environments, API and code can change at any time without notice.

Reviewer: ```suggestion
- The docker provider, including its APIs, should only be used for development/testing purposes. Its API and code can change at any time without notice.
```
